pkg/service/node.go

```diff
@@ -218,15 +218,16 @@ func getDeviceInfo(device string) (string, error) {
 }
 
 func makeFS(device string, fsType string) error {
-	// caution, use -F to force creating the filesystem if it doesn't exit. May not be portable for fs other
-	// than ext family
+	// caution, use force flag when creating the filesystem if it doesn't exit.
 	klog.Infof("Mounting device %s, with FS %s", device, fsType)
-	var force string
+
+	var cmd *exec.Cmd
 	if strings.HasPrefix(fsType, "ext") {
-		force = "-F"
+		cmd = exec.Command("mkfs", "-F", "-t", fsType, device)
+	} else if strings.HasPrefix(fsType, "xfs") {
+		cmd = exec.Command("mkfs", "-t", fsType, "-f", device)
```

Comment:
what if non of them is true?

Reply:
I wonder if we should bail as it's not really supported or tested

Reply:
yes lets error and say exactly what should supported